scipy/_lib/tests/test_linear_assignment.py

```diff
@@ -0,0 +1,98 @@
+from itertools import product
+
+from numpy.testing import assert_array_equal
+import numpy as np
+
+from scipy.optimize import linear_sum_assignment
+from scipy.sparse import csr_matrix, random
+from scipy.sparse.csgraph import min_weight_full_bipartite_matching
+
+
+# Tests that combine scipy.optimize.linear_sum_assignment and
+# scipy.sparse.csgraph.min_weight_full_bipartite_matching
+def test_two_methods_give_expected_result_on_small_inputs():
+    solvers = [(linear_sum_assignment, np.array),
+               (min_weight_full_bipartite_matching, csr_matrix)]
```

Comment:
We could also use pytest parametrization here, although people do push back on that idea sometimes and the test would likely be complex to refactor for that.

I guess because this is kind of copy-pasted from previous test code that wasn't parametrized that could be done later, if at all, to reduce the danger of losing part of the original test.

Reply:
Unpacking the tuples inside the method body, the parametrization could look like

```python
@pytest.mark.parametrize('solver_type,sign,test_case', product(
    [(linear_sum_assignment, np.array),
     (min_weight_full_bipartite_matching, csr_matrix)],
    [-1, 1],
    [
        # Square
        ([[400, 150, 400],
          [400, 450, 600],
          [300, 225, 300]],
         [150, 400, 300]),

        # Rectangular variant
        ([[400, 150, 400, 1],
          [400, 450, 600, 2],
          [300, 225, 300, 3]],
         [150, 2, 300]),

        ([[10, 10, 8],
          [9, 8, 1],
          [9, 7, 4]],
         [10, 1, 7]),

        # Square
        ([[10, 10, 8, 11],
          [9, 8, 1, 1],
          [9, 7, 4, 10]],
         [10, 1, 4]),

        # Rectangular variant
        ([[10, float("inf"), float("inf")],
          [float("inf"), float("inf"), 1],
          [float("inf"), 7, float("inf")]],
         [10, 1, 7]),
    ])
)
def test_two_methods_give_expected_result_on_small_inputs(solver_type, sign, test_case):
    solver, array_type = solver_type
    cost_matrix, expected_cost = test_case
    ...
```
with the method body itself being a good deal simpler than before. Certainly no worse than before.